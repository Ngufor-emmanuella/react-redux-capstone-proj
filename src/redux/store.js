import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import townReducer from './reducers/town';
import countriesReducer from './reducers/country';
import polsReducer from './reducers/polute';

const finalReducer = combineReducers({
  townReducer,
  countriesReducer,
  polsReducer,
});

const store = configureStore({
  reducer: finalReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false,
  }),
});

export default store;
