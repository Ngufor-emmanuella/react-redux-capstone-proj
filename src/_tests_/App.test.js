import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import City from '../components/city';
import Home from '../components/home';
import Polls from '../components/polution';

it('App', () => {
  const utils = render(
    <Provider store={store}>
      <Router>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/city" element={<City />} />
          <Route path="/pollutiondata" element={<Polls />} />
        </Routes>
      </Router>
    </Provider>,
  );
  expect(utils).toMatchSnapshot();
});
