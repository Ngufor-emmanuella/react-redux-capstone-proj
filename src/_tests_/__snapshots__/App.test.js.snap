// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="house"
      >
        <div
          class="search-box"
        >
          <input
            class="search"
            placeholder="Name of country"
            type="text"
            value=""
          />
          <button
            class="btn btn-secondary"
            type="button"
          >
            Search
          </button>
        </div>
        <div
          class="title-home"
        >
          <header>
            <div
              class="map"
            >
              <img
                alt="African map"
                src="./africa.png"
              />
            </div>
            <div
              class="map-map"
            >
              <h2>
                Quality of air in
                <br />
                Africa.
              </h2>
            </div>
          </header>
        </div>
        <div
          class="countries-con container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="house"
    >
      <div
        class="search-box"
      >
        <input
          class="search"
          placeholder="Name of country"
          type="text"
          value=""
        />
        <button
          class="btn btn-secondary"
          type="button"
        >
          Search
        </button>
      </div>
      <div
        class="title-home"
      >
        <header>
          <div
            class="map"
          >
            <img
              alt="African map"
              src="./africa.png"
            />
          </div>
          <div
            class="map-map"
          >
            <h2>
              Quality of air in
              <br />
              Africa.
            </h2>
          </div>
        </header>
      </div>
      <div
        class="countries-con container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
